### PR TITLE
Add proxy config annotation

### PR DESCRIPTION
**Pull request by howardjohn**
This is already implemented in istio/istio. No problem changing the
syntax if people have concerns as it has not yet shipped - just need to make sure it makes it
into 1.6.

### CODE REVIEW

File: annotation/annotations.yaml [hunk 1]
```diff
@@ -343,3 +343,10 @@ annotations:
     hidden: false
     resources:
       - Any
+  - name: istio.io/proxyConfig
```
Comment (rshriram) [inline annotation/annotations.yaml:346]: sidecar.istio.io

Comment (howardjohn) [reply annotation/annotations.yaml:346]: this applies to gateways as well, should we still use sidecar annotation?

Comment (rshriram) [reply annotation/annotations.yaml:346]: maybe call it as proxy.istio.io?

Comment (howardjohn) [reply annotation/annotations.yaml]: @rshriram I updated to proxy.istio.io/config

**Pull request merged**